### PR TITLE
Create role entity

### LINKED ISSUE #204: Create `Role` entity
### Description
Create `Role` entity

### Requirements
- project should be runnable

### Deliverables
- `Role` entity, `RoleType` enum
- `UserRoles` many-to-many linking entity
- `RoleConfiguration` entity configuration
- `UserRolesConfiguration` entity configuration
- `RoleService` foundation service
- add migrations
- register all necessary infrastructure to DI

### PR DESCRIPTION
**Pull request by sulton-max**
Closes #204

### CODE REVIEW

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json [hunk 1]
```diff
@@ -22,8 +22,8 @@
     "SlidingExpirationInSeconds": 300
   },
   "ConnectionStrings": {
-    "DefaultConnection": "Host=localhost;Port=5432;Database=Training.AirBnb.Clone;Username=postgres;Password=postgres",
-    "NotificationConnection": "Host=localhost;Port=5432;Database=Training.AirBnB.Notifications;Username=postgres;Password=postgres",
+    "DefaultConnection": "Host=localhost;Port=5432;Database=Training.AirBnb.Clone;Username=postgres;Password=java2001",
+    "NotificationConnection": "Host=localhost;Port=5432;Database=Training.AirBnB.Notifications;Username=postgres;Password=java2001",
     "RedisConnectionString": "localhost:6379"
```
Comment (botirjonabdumannonovv) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json:27]: change the Password from "java2001" to "postgres"

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/RoleService.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    /// </summary>
+    /// <param name="roleType">The type of role to retrieve.</param>
+    /// <param name="asNoTracking">Flag indicating whether to use 'asNoTracking' while querying (default is false).</param>
+    /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+    /// <returns>A ValueTask representing the retrieved role (or null if not found).</returns>
+    public async ValueTask<Role?> GetByTypeAsync(RoleType roleType, bool asNoTracking = false, CancellationToken cancellationToken = default)
+    {
+        // Retrieve the role using the specified role type asynchronously
+        return await _roleRepository.Get(asNoTracking: asNoTracking)
+            .SingleOrDefaultAsync(role => role.Type == roleType, cancellationToken);
```
Comment (abdura52) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/RoleService.cs:36]: First or default ishlatgan yaxshi ))

**Review by botirjonabdumannonovv — CHANGES_REQUESTED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Validators/RoleValidation.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                // Ensure the 'Id' property is not empty when creating a role
+                RuleFor(code => code.Id).NotEqual(Guid.Empty);
+
+                // Ensure the 'Type' property is not empty when creating a role
+                RuleFor(role => role.Type)
+                    .NotEmpty();
+            }
+        );
+    }
+}
```
Comment (abdura52) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Validators/RoleValidation.cs:36]: Documentation yozish kerak edi comment emas

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json
Comment (Jasurbek1223) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/appsettings.Development.json:1]: Password

File: src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/Identity/Services/IRoleService.cs
Comment (Jasurbek1223) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/Identity/Services/IRoleService.cs:1]: Namespace

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/Interfaces/IRoleRepository.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+/// <summary>
+/// This interface defines the contract for interacting with Role entities in the repository.
+/// </summary>
+public interface IRoleRepository
+{
+    // Retrieves IQueryable collection of Role entities based on the specified predicate.
+    // - 'predicate': Optional filter condition expressed as a lambda expression.
+    // - 'asNoTracking': Indicates whether to track changes (default is false).
+    IQueryable<Role> Get(Expression<Func<Role, bool>>? predicate = default, bool asNoTracking = false);
+}
```
Comment (abdura52) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/Interfaces/IRoleRepository.cs:14]: Bu yerda ham documentation yozish kerak

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/RoleRepository.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        // Initializes the RoleRepository with required dependencies.
+    }
+
+    // Overrides the Get method from the base class to retrieve Role entities.
+    public new IQueryable<Role> Get(Expression<Func<Role, bool>>? predicate, bool asNoTracking)
+    {
+        // Calls the base class's Get method to retrieve Role entities based on provided parameters.
+        return base.Get(predicate, asNoTracking);
+    }
+}
```
Comment (abdura52) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/RoleRepository.cs:26]: commentlarni to'girlash kerak

**Review by abdura52 — APPROVED** [review]

**Review by Jasurbek1223 — APPROVED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by Jasurbek1223 — APPROVED** [review]

**Pull request merged**